pkgs/tools/networking/wsl-vpnkit/default.nix

```diff
@@ -0,0 +1,57 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, makeWrapper
+
+# Runtime dependencies
+, gvproxy
+, dnsutils
+, gawk
+, iproute2
+, iptables
+, iputils
+, wget
+}:
+
+let
+  version = "0.4.1";
+  gvproxyCross = gvproxy.overrideAttrs (_: {
+    buildPhase = "make cross qemu-wrapper vm";
+  });
+in stdenv.mkDerivation {
+  pname = "wsl-vpnkit";
+  inherit version;
+
+  src = fetchFromGitHub {
+    owner = "sakai135";
+    repo = "wsl-vpnkit";
+    rev = "v${version}";
+    hash = "sha256-Igbr3L2W32s4uBepllSz07bkbI3qwAKMZkBrXLqGrGA=";
+  };
+
+  nativeBuildInputs = [ makeWrapper ];
+
+  postPatch = ''
+    substituteInPlace wsl-vpnkit \
+      --replace "/app/wsl-vm" "${gvproxyCross}/bin/vm" \
+      --replace "/app/wsl-gvproxy.exe" "${gvproxyCross}/bin/gvproxy-windows.exe"
+  '';
+
+  installPhase = ''
+    mkdir -p $out/bin
+    cp wsl-vpnkit $out/bin
+  '';
+
+  postFixup = ''
+    wrapProgram "$out/bin/wsl-vpnkit" \
+      --prefix PATH : "${lib.makeBinPath [ dnsutils gawk iproute2 iptables iputils wget ]}"
```

Comment:
Can we use reshole for this? Maybe easier to keep track of all the packages.

Reply:
I tried taking a stab at this but I know too little about resholve to make the experience smooth. I couldn't get it to work, although it did detect a few missing things. It is however much more complicated and I couldn't figure out how to do the VMEXEC_PATH and GVPROXY_PATH commands which needs to be injected. If I set them resholve complained about "Unexpected absolute command path.".

This is what I have so far:
```nix
{ resholve,
, fetchFromGitHub
# Runtime
, coreutils
, dnsutils
, gawk
, gnugrep
, gvproxy
, iproute2
, iptables
, iputils
, wget
}:

let
  gvproxyWin = gvproxy.overrideAttrs (_: {
    buildPhase = ''
      GOARCH=amd64 GOOS=windows go build -ldflags '-s -w' -o bin/gvproxy-windows.exe ./cmd/gvproxy
    '';
  });
in

resholve.mkDerivation rec {
  pname = "wsl-vpnkit";
  version = "0.4.1";

  src = fetchFromGitHub {
    owner = "sakai135";
    repo = "wsl-vpnkit";
    rev = "v${version}";
    sha256 = "sha256-Igbr3L2W32s4uBepllSz07bkbI3qwAKMZkBrXLqGrGA=";
  };

  installPhase = ''
    mkdir -p $out/bin
    cp wsl-vpnkit $out/bin
  '';

  solutions.wsl-vpnkit = {
    scripts = [ "bin/wsl-vpnkit" ];
    interpreter = "none";
    inputs = [
      coreutils
      dnsutils
      gawk
      gnugrep
      iproute2
      iptables
      iputils
      wget
    ];
    fix = {
      "$VMEXEC_PATH" = [ "${gvproxy}/bin/vm" ];
      "$GVPROXY_PATH" = [ "${gvproxyWin}/bin/gvproxy-windows.exe" ];
    };
    execer = [
      "cannot:${iproute2}/bin/ip"
      "cannot:${wget}/bin/wget"
    ];
    fake.external = [ "ping" ];
  };
}
```

Reply:
Okay, keeping the variable as substitution worked, what do you think?
```nix
{ resholve
, fetchFromGitHub
, # Runtime
  coreutils
, dnsutils
, gawk
, gnugrep
, gvproxy
, iproute2
, iptables
, iputils
, wget
,
}:

let
  gvproxyWin = gvproxy.overrideAttrs (_: {
    buildPhase = ''
      GOARCH=amd64 GOOS=windows go build -ldflags '-s -w' -o bin/gvproxy-windows.exe ./cmd/gvproxy
    '';
  });
in
resholve.mkDerivation rec {
  pname = "wsl-vpnkit";
  version = "0.4.1";

  src = fetchFromGitHub {
    owner = "sakai135";
    repo = "wsl-vpnkit";
    rev = "v${version}";
    sha256 = "sha256-Igbr3L2W32s4uBepllSz07bkbI3qwAKMZkBrXLqGrGA=";
  };

  postPatch = ''
    substituteInPlace wsl-vpnkit \
      --replace "/app/wsl-vm" "${gvproxy}/bin/vm" \
      --replace "/app/wsl-gvproxy.exe" "${gvproxyWin}/bin/gvproxy-windows.exe"
  '';

  installPhase = ''
    mkdir -p $out/bin
    cp wsl-vpnkit $out/bin
  '';

  solutions.wsl-vpnkit = {
    scripts = [ "bin/wsl-vpnkit" ];
    interpreter = "none";
    inputs = [
      gvproxy
      gvproxyWin

      coreutils
      dnsutils
      gawk
      gnugrep
      gvproxy
      gvproxyWin
      iproute2
      iptables
      iputils
      wget
    ];

    keep = {
      "$VMEXEC_PATH" = true;
      "$GVPROXY_PATH" = true;
    };

    execer = [
      "cannot:${iproute2}/bin/ip"
      "cannot:${wget}/bin/wget"
    ];
    fake.external = [ "ping" ];
  };
}
```

Reply:
That looks awesome! Please commit that

Reply:
I have rebased with these changes now. Seems the latest version of `gvproxy` also renamed vm to `gvforwarder`.